web-app/src/main/webapp/script.js

```diff
@@ -12,17 +12,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function initMap() {
+  const map = new google.maps.Map(document.getElementById("map"), {
+    center: { lat: -34.397, lng: 150.644 },
+    zoom: 5,
+  });
+  const infoWindow = new google.maps.InfoWindow();
+  displayUserLocation(map, infoWindow);
+}
+
+function displayUserLocation(map, infoWindow) {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(
+      function(position) {
+        let userPosition = {
+          lat: position.coords.latitude,
+          lng: position.coords.longitude,
+        };
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+        const marker = new google.maps.Marker({
+          position: userPosition,
+          map: map,
+        });
+        map.setCenter(userPosition);
+      },
+      function() {
+        handleLocationError(true, infoWindow, map.getCenter());
+      }
+    );
+  } else {
+    // Browser doesn't support Geolocation
+    handleLocationError(false, infoWindow, map.getCenter());
+  }
+}
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+function handleLocationError(browserHasGeolocation, infoWindow, pos) {
```

Comment:
nit: split this into two functions instead of having a boolean toggle between the two messages. Alternatively, change this to "showInfoWindow" and let the caller specify the string message.